lodash_exercise/lodash.js

```diff
@@ -1,99 +1,150 @@
 }
 
-function take(){
-
+function fromPairs(array) {
+  return array.reduce(function(prev, curr, idx) {
+    prev[curr[0]] = curr[1];
+    return prev;
+  }, {});
+  // var index = -1;
+  // var l = array == null ? 0 : array.length;
+  // var result = {};
+  // while (++index < l) {
+  //   var pair = array[index];
+  //   result[(pair[0] = pair[1])];
+  // }
+  // return result;
 }
 
-function takeRight(){
-
+function head(array) {
+  return array[0];
 }
 
-function union(){
-
+function take(array, n = 1) {
+  return array.slice(0, n);
 }
 
-function zipObject(){
-
+function takeRight(array, n = 1) {
+  return n == 0 ? [] : array.slice(-n);
 }
 
-function includes(){
-
+function union(...array) {
+  return [...new Set(array[0].concat(array[1]))];
 }
 
-function sample(){
-
+function zipObject(arr, arr1) {
+  return arr.reduce(function(prev, curr, idx) {
+    prev[curr] = arr1[idx];
+    return prev;
+  }, {});
 }
 
-function cloneDeep(){
+function includes(collection, target, startingIndex = 0) {
+  if (typeof collection === 'string') {
+    return collection.includes(target, collection.slice(startingIndex));
+  }
+  collection = Array.isArray(collection)
+    ? collection
+    : Object.values(collection);
 
+  return collection.includes(target, drop(collection, startingIndex));
 }
 
-function sumBy(){
-
+function sample(array) {
+  return Math.round(Math.random(...array) + array[0]);
 }
+//[{ a: 1 }, { b: 2 }]
+function cloneDeep(array) {}
 
-function inRange(){
-
+function sumBy(array, iteratee) {
+  return array && array.length ? iteratee(identity) : 0;
 }
 
-function has(){
-
+function inRange(num, start, end) {
+  num = Math.floor(num);
+  if (end == undefined) {
+    start = 0;
+  }
+  if (num < 0) {
+    return num <= start && num >= end ? true : false;
+  } else {
+    return num >= start && num <= end ? true : false;
+  }
 }
 
-function omit(){
-
+function has(object, other) {
+  return object.hasOwnProperty(...other) ? true : false;
 }
 
-function pick(){
+function omit(obj, arr) {}
 
+function pick(obj, arr) {
+  return arr.reduce(function(newObj, key) {
+    if (key in obj) {
+      newObj[key] = obj[key];
+    }
+    return newObj;
+  }, {});
 }
 
-function pickBy(){
+function pickBy(obj, cb) {}
 
-}
-
-function omitBy(){
+function omitBy() {}
 
+function padEnd(str, target, char) {
+  return str.padEnd(target, char);
 }
 
-function padEnd(){
-
+function repeat(str, n) {
+  return str.repeat(n);
 }
 
-function repeat(){
-
+function upperFirst(str) {
+  return str[0].toUpperCase() + str.slice(1);
 }
 
-function upperFirst(str){
-
+function flatten(array) {
+  var flat = [];
+  for (var i = 0; i < array.length; i++) {
+    if (Array.isArray(array[i])) {
+      flat = flat.concat(array[i]);
+    } else {
+      flat.push(array[i]);
+    }
+  }
+  return flat;
 }
 
-function flatten(){
+function zip() {}
 
-}
-
-function zip(){
+function unzip() {}
 
+function flip() {
+  return [...arguments].reduceRight(function(prev, curr) {
```

Comment:
read the instructions for this function. Flip should return a function to you with the arguments in reverse